lib/git_local/repository.rb

```diff
@@ -104,5 +109,10 @@ def reset_to_latest_from_origin
     def repo_path
       @repo_path ||= "#{local_directory}/#{org_repo}"
     end
+
+    def valid_protocol(protocol)
+      raise InvalidProtocol, "Protocol must be either HTTPS or SSH." unless ["SSH", "HTTPS"].include?(protocol)
```

Comment:
"SSH" and "HTTPS" might be better expressed in a constant like:

```ruby
VALID_PROTOCOLS = {
  ssh: "SSH",
  https: "HTTPS"
}.freeze
```

Then we would use `VALID_PROTOCOLS[:ssh]` etc.